### ISSUE TITLE
Support "initial credentials" in `universal` case

### ISSUE THREAD

**Issue by yskopets** [Kong/kuma#287]
### Summary

* Support "initial credentials" in `universal` case (similarly to `Service Account token` on `k8s`)

**Comment by jakubdyszkiewicz** [comment]
# Design

When connecting to the SDS server, a Dataplane has to prove its identity. Control Plane will distribute a JWT token which contains name, mesh and IP address. This token will be validated on the Control Plane side.

Control Plane will expose extra server that generates JWT tokens. This server can be exposed in two ways:
1) non secure plaintext HTTP only running in localhost.
This is useful for two reasons. One is demo purposes. Second is enable administrator to write a script that generates token and put them in some secured storage of their choice.

2) secured HTTPS visible from the outside. It also has to verify the identity of machine connecting to it. The verification can be either by basic auth/api key or client TLS certs.
This can provide integration that is not running on the control plane machine.

Both ways are not mutually exclusive. 

Once generated, the token has to be distributed to the machine. This may vary on the infrastructure running at given company.
Use case with Puppet:
Puppet Master can retrieve Initial Token from the Control Plane using client certs (or basic auth/api key) that is stored in config of the Control Plane. Initial Token is 
then stored in defined (in Puppet class/resource) path.

_TODO: investigate more how it can be used with other automation tools like Ansible or Chef._

Path to the Initial Token has to be read by the Control Plane, because it's a part of CDS/LDS config. There are two ways how to pass it:
1) `kuma-dp run -initialTokenPath=/path.to.file`. Then we send the path as a part of bootstrap request for bootstrap Envoy configuration. This path is stored as a part of Envoy Metadata and is carried over with every XDS request.
2) Store it in Dataplane resource.
```
type: Dataplane
name: dp-1
mesh: default
initialTokenPath: /path.to.file
networking:
  inbound:
...
```
and just `kuma-dp run`
Then on the Control Plane side we just read it from the Resource Store.

Although second option is easier to implement, we already have binary path and config dir which are parts of CLI args and not the Dataplane resource.

**Comment by yskopets** [comment]
> Control Plane will distribute a JWT token

`Control Plane` cannot distribute JWT tokens, only provide API to generate them

**Comment by yskopets** [comment]
> This token will be validated on the Control Plane side

Can be more specific - this token will be validated by `SDS` service before issuing an mTLS cert for  a requesting workload

**Comment by yskopets** [comment]
Suggestions:
* add an overview section (right now, the proposal documents all technical solutions that will be implemented, but it's not clear how they all fit together)
* describe how TLS cert and list of allowed clients for the `initial token endpoint` will be managed
* give an example how the updated `Getting Started` flow will look like
* does it make sense to add a new command to `kumactl` to help admins to generate tokens  ? E.g., `kumactl generate initial-token ...`

**Comment by subnetmarco** [comment]
To summarize, in order to generate a token:

```sh
$ kumactl generate dp-token --name {NAME} --mesh {MESH}
TOKEN_OUTPUT
```
This will output the token. It can be stored in a file with `$ ... > token`.

**This command can only be executed on the same machine where `kuma-cp` is running.**

Once the token has been generated, we can then pass it to the `kuma-dp` process when executing the `run` command, like:

```sh
$ kuma-dp run --token-path /path/to/token
```

**Comment by jakubdyszkiewicz** [comment]
Implemented in 0.3.0
